source/ada/lsp-ada_documents.adb

```diff
@@ -2391,12 +2414,40 @@ package body LSP.Ada_Documents is
          return;
       end if;
 
-      --  Get keyword completion
-      if not In_End_Label then
-         Get_Keywords_Completion
-           (Node   => Node,
-            Prefix => Prefix,
-            Result => Result);
+      declare
+         Previous_Char : constant String :=
+           (if Position.character > 1 then
+               Self.Get_Text_At
+              (Start_Pos => LSP.Messages.Position'
+                   (line      => Position.line,
+                    character => Position.character - 2),
+               End_Pos   => LSP.Messages.Position'
+                 (line      => Position.line,
+                  character => Position.character - 2))
+            else
+               "");
+      begin
+         --  Propose keyword completion if we are not within and end label
+         --  and if there is no previous character of if it's a whitespace (we
+         --  don't want to propose keywords after typing '(' to feed subprogram
+         --  parameters for instance).
```

Comment:
some keywords are valid after (, for instance in

     S : String (1 .. 10) := (others => 'a');